litestar/_openapi/plugin.py

```diff
@@ -1,22 +1,56 @@
 from __future__ import annotations
 
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Any, Dict
 
 from litestar._openapi.datastructures import OpenAPIContext
 from litestar._openapi.path_item import create_path_item_for_route
-from litestar.exceptions import ImproperlyConfiguredException
+from litestar.di import Provide
+from litestar.enums import MediaType
+from litestar.exceptions import ImproperlyConfiguredException, NotFoundException
+from litestar.handlers import get
+from litestar.openapi.plugins import JsonRenderPlugin
 from litestar.plugins import InitPluginProtocol
 from litestar.plugins.base import ReceiveRoutePlugin
+from litestar.response import Response
+from litestar.router import Router
 from litestar.routes import HTTPRoute
+from litestar.status_codes import HTTP_404_NOT_FOUND
 
 if TYPE_CHECKING:
     from litestar.app import Litestar
     from litestar.config.app import AppConfig
+    from litestar.connection import Request
+    from litestar.handlers import HTTPRouteHandler
     from litestar.openapi.config import OpenAPIConfig
+    from litestar.openapi.plugins import OpenAPIRenderPlugin
     from litestar.openapi.spec import OpenAPI
     from litestar.routes import BaseRoute
 
 
+def handle_schema_path_not_found(path: str = "/") -> Response:
+    """Handler for returning HTML formatted errors from not-found schema paths.
+
+    This preserves backward compatibility with the Controller-based OpenAPI implementation.
+    """
+    if path.endswith((".json", ".yaml", ".yml")):
+        raise NotFoundException
+
+    content = b"""
+    <!DOCTYPE html>
+    <html>
+        <head>
+            <title>404 Not found</title>
+            <meta charset="utf-8"/>
+            <meta name="viewport" content="width=device-width, initial-scale=1">
+        </head>
+        <body>
+            <h1>Error 404</h1>
+        </body>
+    </html>
+    """
+    return Response(content, media_type=MediaType.HTML, status_code=HTTP_404_NOT_FOUND)
```

Comment:
Consider adding this handler to the openapi config object so that it can be overridden by users

Reply:
This can be an improvement if requested.